src/time-picker/time-picker.jsx

```diff
@@ -159,15 +198,19 @@ const TimePicker = React.createClass({
   },
 
   openDialog() {
-    this.setState({
-      dialogTime: this.getTime(),
-    });
+    if (this.state.time) {
```

Comment:
The Clock component has a `initialTime` property that sets its `selectedTime` state, if I let null pass here,  the following line will https://github.com/callemall/material-ui/blob/master/src/time-picker/clock.jsx#L58, throw a `TypeError`.

Another option could be :

`this.setState({ dialogTime: this.state.time || new Date() })`

Which could have more sense, since it will update the clock time.